src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -911,10 +915,70 @@ protected override void AddExistingDocument(DocumentId documentId, string filePa
                 throw new Exception(ServicesVSResources.Could_not_find_location_of_folder_on_disk);
             }
 
-            return AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+            AddDocumentToProjectItems(folder.ProjectItems, documentId, folderPath, documentName, initialText, filePath, documentKind);
+        }
+
+        private bool TryAddEditorConfigToSolutionItems(
+            DocumentInfo documentInfo,
+            SourceText text)
+        {
+            // We're going to see if this is an .editorconfig being added at the solution level, and if so process it specially; so verify first that it is one
+            if (CurrentSolution.FilePath is null || documentInfo.FilePath is null)
+                return false;
+
+            if (PathUtilities.GetFileName(documentInfo.FilePath) != ".editorconfig")
```

Comment:
i'm ok with this.  i just want to chekc that we've considered the case sensitivity implications of this.

Reply:
This is just matching the generated file we create here: https://github.com/dotnet/roslyn/blob/89875d733146fa149f725aefdcfcda8776680417/src/Features/Core/Portable/CodeFixes/Configuration/ConfigurationUpdater.cs#L340

Reply:
And to explain a bit further: this path isn't taken for a file on disk, but rather a new file being created. We don't expect folks to be doing something different there, and if they were to, worst case scenario, they just take the other (existing) path that already has been active for a long time.